plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -145,6 +145,142 @@ func NoDiskConflict(pod *api.Pod, existingPods []*api.Pod, node string) (bool, e
 	return true, nil
 }
 
+type MaxPDVolumeCountChecker struct {
+	filter     PDVolumeFilter
+	maxVolumes uint
+	pvInfo     PersistentVolumeInfo
+	pvcInfo    PersistentVolumeClaimInfo
+}
+
+// PDVolumeFilter contains information on how to filter PD Volumes when checking PD Volume caps
+type PDVolumeFilter struct {
+	// Filter normal volumes
+	FilterVolume           func(vol *api.Volume) (id string, relevant bool)
+	FilterPersistentVolume func(pv *api.PersistentVolume) (id string, relevant bool)
+}
+
+// NewMaxPDVolumeCountPredicate creates a predicate which evaluates whether a pod can fit based on the
+// number of volumes which match a filter that it requests, and those that are already present.  The
+// maximum number is configurable; Amazon recommends no more than 40 EBS volumes, while GCE reccomends
```

Comment:
I would drop the part of this sentence after the semicolon because it's just more places you'll have to update when this number changes, and also the constant is defined in a different file.